File: frontend/src/types/Lead.ts
// types/Lead.ts
export interface LeadResponse {
  id: string;
  first_name: string;
  last_name: string;
  email: string;
  phone?: string;
  company?: string;
  status: string;
  priority?: string;
  source?: string;
  created_at: string;
  updated_at: string;
}

export interface Lead {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  company?: string;
  status: string;
  priority?: string;
  source?: string;
  createdAt: string;
  updatedAt: string;
}

export interface LeadFormState {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
  company?: string;
  status: string;
  priority?: string;
  source?: string;
}

File: backend/src/main/java/com/realestatecrm/entity/Lead.java
package com.realestatecrm.entity;

import jakarta.persistence.*;
import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "leads")
public class Lead {
    @Id
    @GeneratedValue
    private UUID id;

    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private String source;

    @Enumerated(EnumType.STRING)
    private Status status = Status.NEW;

    public enum Status { NEW, CONTACTED, QUALIFIED, OPPORTUNITY, CONTRACT, CLOSED_WON, CLOSED_LOST }

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_to")
    private User assignedTo;

    private Instant createdAt = Instant.now();
    private Instant updatedAt = Instant.now();

    // getters / setters (generated minimally)
    public UUID getId() { return id; }
    public String getFirstName() { return firstName; }
    public void setFirstName(String firstName) { this.firstName = firstName; }
    public String getLastName() { return lastName; }
    public void setLastName(String lastName) { this.lastName = lastName; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
    public String getSource() { return source; }
    public void setSource(String source) { this.source = source; }
    public Status getStatus() { return status; }
    public void setStatus(Status status) { this.status = status; }
    public User getAssignedTo() { return assignedTo; }
    public void setAssignedTo(User assignedTo) { this.assignedTo = assignedTo; }
    public Instant getCreatedAt() { return createdAt; }
    public Instant getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(Instant updatedAt) { this.updatedAt = updatedAt; }
}
